src/cli/cmd/apply/mod.rs

```diff
@@ -0,0 +1,200 @@
+mod home_manager;
+#[cfg(target_os = "macos")]
+mod nix_darwin;
+#[cfg(target_os = "linux")]
+mod nixos;
+
+use std::{
+    os::unix::prelude::PermissionsExt,
+    path::PathBuf,
+    process::{ExitCode, Stdio},
+};
+
+use clap::{Parser, Subcommand};
+use color_eyre::eyre::Context;
+
+use crate::{
+    cli::{
+        cmd::{nix_command, parse_flake_output_ref},
+        error::FhError,
+    },
+    path,
+};
+
+use self::home_manager::{HomeManager, HOME_MANAGER_SCRIPT};
+
+#[cfg(target_os = "linux")]
+use self::nixos::{NixOs, NIXOS_PROFILE, NIXOS_SCRIPT};
+
+#[cfg(target_os = "macos")]
+use self::nix_darwin::{NixDarwin, DARWIN_REBUILD_ACTION, NIX_DARWIN_PROFILE, NIX_DARWIN_SCRIPT};
+
+use super::{CommandExecute, FlakeHubClient};
+
+/// Apply the configuration at the specified FlakeHub output reference to the current system
+#[derive(Parser)]
+pub(crate) struct ApplySubcommand {
+    #[clap(subcommand)]
+    system: System,
+
+    #[clap(from_global)]
+    api_addr: url::Url,
+}
+
+#[derive(Subcommand)]
+enum System {
+    /// Resolve the store path for a Home Manager configuration and run its activation script
+    HomeManager(HomeManager),
+
+    #[cfg(target_os = "macos")]
+    /// Resolve the store path for a nix-darwin configuration and run its activation script
+    NixDarwin(NixDarwin),
+
+    #[cfg(target_os = "linux")]
+    /// Apply the resolved store path on a NixOS system
+    #[clap(name = "nixos")]
+    NixOs(NixOs),
```

Comment:
If we can avoid `cfg(target_os...`, I'd like to. We recently deleted a ton of this from the Nix installer project because it makes developing on different platforms much much harder.